core/src/utils/overlays.ts

```diff
@@ -533,7 +533,17 @@ const focusPreviousElementOnDismiss = async (overlayEl: any) => {
   }
 
   await overlayEl.onDidDismiss();
-  previousElement.focus();
+
+  /**
+   * If the user has already removed focus
+   * from the overlay (For example, focusing
+   * a text box after tapping a button in an
+   * action sheet) then don't restore focus
+   * to previous element
+   */
+  if (document.activeElement === null || document.activeElement === document.body) {
```

Comment:
Can we also add a comment that explains focus is always moved to the body when the overlay is dismissed? Maintainers may not know why we specifically check the body here.

Reply:
I added some MDN references. Apologies if its now a bit too wordy, feel free to make further suggestions!